### PR TITLE
Finished creating tcp connection

### PR DESCRIPTION
**Pull request by eoswald**

### CODE REVIEW

File: bin/main.go [hunk 1]
```diff
@@ -19,13 +20,32 @@ var connect = flag.String("connect", "",
...
 	"Whether we should try and find neighboring routers")
 
-func FindNeighbors(dev net.Interface, key node.PublicKey) <-chan node.NodeAddress {
+func GetLinkLocalAddr(dev net.Interface) (*net.IPAddr, error) {
+	cidr_ll_addr, err := dev.Addrs()
+	if err != nil {
+		return nil, err
+	}
+
+	ll_addr, _, err := net.ParseCIDR(cidr_ll_addr[1].String())
```
Comment (c00w) [inline bin/main.go]: Does this always return a link local address? Is it always the second one?

**Comment by c00w** [comment]
Looks good to me; feel free to merge at will.

File: bin/main.go (hunk 1 above)
Comment (eoswald) [inline bin/main.go]: I tested this in my development VM and my VPS and the second one always appears to be a link local address. Do you want to test it on a couple machines before I merge?

Comment (eoswald) [inline bin/main.go]: Well specifically the second one is always an ipv6 link local. The first one is always ipv4.

File: bin/main.go [hunk 2]
```diff
@@ -19,13 +20,42 @@ var connect = flag.String("connect", "",
...
+	dev_addrs, err := dev.Addrs()
+	if err != nil {
+		return nil, err
+	}
+
+	var cidr_ll_addr net.Addr
+
+	for _, addr := range dev_addrs {
+		prefix := addr.String()[:3]
+		if prefix == "fe8" || prefix == "fe9" || prefix == "fea" || prefix == "feb" {
```
Comment (c00w) [inline bin/main.go]: FYI doing this with bit math is slightly cleaner.

File: bin/main.go [hunk 3]
```diff
@@ -19,13 +20,42 @@ var connect = flag.String("connect", "",
...
+			cidr_ll_addr = addr
+			break
+		}
+	}
+
+	ll_addr, _, err := net.ParseCIDR(cidr_ll_addr.String())
```
Comment (c00w) [inline bin/main.go]: Should probably add a log.Fatal or return an error if you can't find it.

File: neighbor_finder_test.go [hunk 4]
```diff
@@ -45,8 +46,13 @@ func TestBasicExchange(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	nf1 := NewNeighborData(pk1)
-	nf2 := NewNeighborData(pk2)
+	link_local_addr, err := net.ResolveIPAddr("ip6", "fe80::%0")
```
Comment (c00w) [inline neighbor_finder_test.go]: Is zone0 always local host? if so add a comment?

Comment (eoswald) [inline neighbor_finder_test.go]: It's just a test zone since our neighbor finder tests don't use networking devices at the moment

**Pull request merged**